packages/validators/test/v2023_09_28/experienceValidator.spec.ts

```diff
@@ -0,0 +1,708 @@
+          typeof updatedExperience
+        >;
+
+        expect(result.success).toBe(false);
+        expect(result.error.issues[0].message).toBe(
+          'The first breakpoint should include the following attributes: { "id": "desktop", "query": "*" }',
+        );
+      });
+
+      it(`fails if breakpoints are not ordered from largest to smallest`, () => {
+        const breakpoints = [
+          { id: 'desktop', query: '*', previewSize: 'large', displayName: 'Desktop' },
+          { id: 'mobile', query: '<768px', previewSize: 'small', displayName: 'Mobile' },
+          { id: 'tablet', query: '<1024px', previewSize: 'medium', displayName: 'Tablet' },
+        ];
+        const componentTree = experience.fields.componentTree[locale];
+        const updatedExperience = {
+          ...experience,
+          fields: {
+            ...experience.fields,
+            componentTree: { [locale]: { ...componentTree, breakpoints } },
+          },
+        };
+        const result = validateExperienceFields(updatedExperience, schemaVersion) as SafeParseError<
+          typeof updatedExperience
+        >;
+
+        expect(result.success).toBe(false);
+        expect(result.error.issues[0].message).toBe(
+          'Breakpoints should be ordered from largest to smallest pixel value',
+        );
+      });
+    });
+
+    describe('schemaVersion', () => {
+      it(`fails if schemaVersion is missing`, () => {
+        const componentTree = experience.fields.componentTree[locale];
+        const updatedExperience = {
+          ...experience,
+          fields: {
+            ...experience.fields,
+            componentTree: { [locale]: { ...componentTree, schemaVersion: undefined } },
+          },
+        };
+        const result = validateExperienceFields(updatedExperience, schemaVersion) as SafeParseError<
+          typeof updatedExperience
+        >;
+
+        expect(result.success).toBe(false);
+        expect(result.error.issues[0].message).toBe('Invalid input');
+      });
+    });
+
+    describe('children', () => {
+      it(`fails if children is not an array`, () => {
+        const componentTree = experience.fields.componentTree[locale];
+        const updatedExperience = {
+          ...experience,
+          fields: {
+            ...experience.fields,
+            componentTree: { [locale]: { ...componentTree, children: {} } },
+          },
+        };
+        const result = validateExperienceFields(updatedExperience, schemaVersion) as SafeParseError<
+          typeof updatedExperience
+        >;
+
+        expect(result.success).toBe(false);
+        expect(result.error.issues[0].message).toBe('Expected array, received object');
+      });
+
+      (
+        [
+          ['definitionId', 'string'],
+          ['variables', 'object'],
+          ['children', 'array'],
+        ] as const
+      ).map(([attribute, type]) =>
+        it(`fails if '${attribute}' is missing`, () => {
```

Comment:
In jest, such lists of test are usually defined via `it.each` - see the docs [here](https://jestjs.io/docs/api#testeachtablename-fn-timeout).